src/main/target/WORMFC/target.h

```diff
@@ -0,0 +1,140 @@
+ */
+
+#pragma once
+
+#define TARGET_BOARD_IDENTIFIER "RRF4"
+
+#define USBD_PRODUCT_STRING     "Worm FC"
+
+#define LED0_PIN    				PA15
+#define LED1_PIN					PC14
+
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+
+#define BEEPER                  	PB14
+#define BEEPER_INVERTED
+
+#define INVERTER_PIN_UART6      	PB13 // PC0 used as inverter select GPIO
+#define INVERTER_PIN_UART3      	PB12 // PC0 used as inverter select GPIO
+
+// MPU6500 interrupt
+#define USE_EXTI
+#define MPU_INT_EXTI            	PC4
+#define USE_MPU_DATA_READY_SIGNAL
+//#define DEBUG_MPU_DATA_READY_INTERRUPT
+
+#define ICM20689_CS_PIN          PA4
+#define ICM20689_SPI_INSTANCE    SPI1
+
+#define MPU6500_CS_PIN          	PA4
+#define MPU6500_SPI_INSTANCE    	SPI1
+
+// ACC section -- start
+#define ACC
+
+#define USE_ACC_SPI_ICM20689
+#define ACC_ICM20689_ALIGN       CW180_DEG_FLIP
+
+#define USE_ACC_MPU6500
+#define USE_ACC_SPI_MPU6500
+#define ACC_MPU6500_ALIGN       CW180_DEG_FLIP
+// ACC section -- end
+
+// GYRO section -- start
+#define GYRO
+
+#define USE_GYRO_SPI_ICM20689
+#define GYRO_ICM20689_ALIGN      CW180_DEG_FLIP
+
+#define USE_GYRO_MPU6500
+#define USE_GYRO_SPI_MPU6500
+#define GYRO_MPU6500_ALIGN      CW180_DEG_FLIP
+// GYRO section -- end
+
+#define USE_VCP
+#define VBUS_SENSING_PIN        PA9
+#define VBUS_SENSING_ENABLED
+
+#define USE_UART1
+#define UART1_RX_PIN            PB7
+#define UART1_TX_PIN            PB6
+
+#define USE_UART3
+#define UART3_RX_PIN            PB11
+#define UART3_TX_PIN            PB10
+
+#define USE_UART6
+#define UART6_RX_PIN            PC7
+#define UART6_TX_PIN            PC6
+
+#define SERIAL_PORT_COUNT       4 //VCP, USART1, USART3, USART6, SOFTSERIAL x 2
+
+#define USE_ESCSERIAL
+#define ESCSERIAL_TIMER_TX_HARDWARE 0 // PWM 1
+
+#define USE_SPI
+
+#define USE_SPI_DEVICE_1
+
+#define USE_SPI_DEVICE_3
+#define SPI3_SCK_PIN            PB3
+#define SPI3_MISO_PIN           PB4
+#define SPI3_MOSI_PIN           PB5
+
+#define USE_OSD
+#define USE_MAX7456
+#define MAX7456_SPI_INSTANCE    SPI3
+#define MAX7456_SPI_CS_PIN      PC0
+#define MAX7456_SPI_CLK         (SPI_CLOCK_STANDARD) // 10MHz
+#define MAX7456_RESTORE_CLK     (SPI_CLOCK_FAST)
+
+#define BARO
+#define USE_BARO
+#define USE_BARO_LPS
+#define LPS_SPI_INSTANCE SPI3
+#define LPS_CS_PIN PB8
+
+// #define MAX7456_DMA_CHANNEL_TX              DMA1_Stream5
+// #define MAX7456_DMA_CHANNEL_RX              DMA1_Stream0
+// #define MAX7456_DMA_IRQ_HANDLER_ID          DMA1_ST0_HANDLER
+
+#define USE_ADC
+#define BOARD_HAS_VOLTAGE_DIVIDER
+#define VBAT_ADC_PIN            PC1
+#define RSSI_ADC_PIN            PC2
+
+#define DEFAULT_RX_FEATURE      FEATURE_RX_SERIAL
+#define SERIALRX_PROVIDER       SERIALRX_SBUS
+
+#define USE_SERIAL_4WAY_BLHELI_INTERFACE
+#define ENABLE_BLACKBOX_LOGGING_ON_SDCARD_BY_DEFAULT
+
+#define USE_SDCARD
+#define USE_SDCARD_SDIO
+#define SDIO_DMA
+//This is not used on SDIO
+#define SDCARD_SPI_CS_PIN PB15
```

Comment:
This can be removed?

Reply:
I need to change stuff in fc_init.c to remove it;)

Reply:
Does the board have the pin connected? If so it should stay defined. If not, you can probably set it to `NONE`.

Reply:
There is no such thing as CS Pin in SDIO interface. I left it like that because otherwise compiler complains and it requires some changes on fc_init. Might do it now or later as you want

Reply:
I know. But users might want to use this board to control the SD card port with SPI (there's mods that use this ports to attach non-sdcard devices, or a bug might surface in the SDIO driver, or simple preference), so there might be value to have the CS pin connected to a spare MCU pin...

If the SD card socket's CS pin is not connected to PB15, then it should not state that, this is confusing. Set it to `NONE` instead.

Reply:
It's either SPI or SDIO, pinout is totally different between using those two. SDIO in 4 Bit wide mode requires a total of 6 pins to talk to SD Card. In a 1Bit wide mode similar to SPI, it needs 3 pins. But still, the pinout is totally different. What I'm saying is none of the current targets will be able to use this, you need to design FC with this in mind basically. However in my opinion, the interface has so many benefits it's worth to force people into switching to it. CS Pin, therefore, is not used at all when using SDIO mode (and won't be any useful), I had no idea you can put NONE instead, will do it right away.

Reply:
Ah, ok, my bad for not realising that this was not going to work with the existing board designs. So I assume there are dedicated pins on every MCU that have to be used for SDIO, and they are _not_ identical to the ones that can also be used for SPI mode? That's a bit of a shame.

Do MCUs allow alternative pins to be used for SDIO, like they do for most other functions? If so, I think we should extend the SDIO driver to allow assignment of the SDIO pins through the config file / configuration (as a move towards board independant targets). In general, I think it makes good sense to have definitions for all used pins in the target definition, since this makes it easier for users to tinker with their boards.

Reply:
I've never encountered a board where you could remap those pins, also from what I've seen so far in L4, F4, F7 they've been assigned to the same ports/pins int the series, therefore as long as you connected your card properly there is no need to have definitions with pins for each target. I think F407-Disc breaks out all necessary pins so you can test it if you want. The only problem is lack of adapters to SDIO on the market so most likely you will have to solder wires...

Reply:
@jflyper: What do you think is the best way to make these pins show as used in `resource list`?

Reply:
```
# resource list
Currently active IO resource assignments:
(reboot to update)
--------------------
A00: FREE
A01: FREE
A02: MOTOR 3
A03: MOTOR 4
A04: MPU_CS
A05: SPI_SCK 1
A06: SPI_MISO 1
A07: SPI_MOSI 1
A08: MOTOR 2
A09: FREE
A10: MOTOR 1
A11: USB
A12: USB
A13: FREE
A14: FREE
A15: LED 1
B00: FREE
B01: FREE
B02: FREE
B03: SPI_SCK 3
B04: SPI_MISO 3
B05: SPI_MOSI 3
B06: FREE
B07: FREE
B08: BARO_CS
B09: FREE
B10: FREE
B11: FREE
B12: INVERTER 3
B13: INVERTER 6
B14: BEEPER
B15: SDCARD
C00: SPI_PREINIT
C01: ADC_BATT
C02: FREE
C03: FREE
C04: MPU_EXTI
C05: FREE
C06: FREE
C07: FREE
C08: SDCARD
C09: SDCARD
C10: SDCARD
C11: SDCARD
C12: SDCARD
C13: FREE
C14: LED 2
C15: FREE
D00: FREE
D01: FREE
D02: SDCARD
D03: FREE
D04: FREE
D05: FREE
D06: FREE
D07: FREE
D08: FREE
D09: FREE
D10: FREE
D11: FREE
D12: FREE
D13: FREE
D14: FREE
D15: FREE

Currently active DMA:
--------------------
DMA1 Stream 0: FREE
DMA1 Stream 1: FREE
DMA1 Stream 2: FREE
DMA1 Stream 3: FREE
DMA1 Stream 4: FREE
DMA1 Stream 5: FREE
DMA1 Stream 6: FREE
DMA1 Stream 7: FREE
DMA2 Stream 0: FREE
DMA2 Stream 1: FREE
DMA2 Stream 2: FREE
DMA2 Stream 3: SDCARD
DMA2 Stream 4: ADC
DMA2 Stream 5: FREE
DMA2 Stream 6: FREE
DMA2 Stream 7: FREE

Use: 'resource' to see how to change resources.
```

This is how it looks like at the moment.... Indeed not very intuitive, what we can maybe do is to point out, command, clock and data pins. Something like SDCARD_CMD, SDCARD_DATA, SDCARD_CLK

Reply:
That looks good enough to me - I did not expect it to show up at all. I think that is good enough to satisfy tinkerers. If the pins cannot be changed then there is little reason for them to change them, but the fact that they don't show up as `FREE` means they won't reassign them to something else (hopefully).

Reply:
Ok great, I'll fix other stuff later in the evening so we can merge this and also #4857. I have some work to do now:)